### PR TITLE
make backups and wiki runs safer

### PR DESCRIPTION
**Pull request by tridge**
- check that each wiki did build ok
- check for key python version and modules

### CODE REVIEW

File: update.py [hunk 1]
```diff
@@ -782,7 +807,7 @@ def copy_static_html_sites(site, destdir):
     )
     parser.add_argument(
         '--destdir',
-        default="/var/sites/wiki/web" if platform.system() != "Windows" else os.path.join(os.path.dirname(__file__), "..", "wiki"),  # noqa: E501
+        default=None,
```
Comment (peterbarker) [inline update.py:810]: Are we supplying this on the server?

Comment (tridge) [reply update.py:807]: yes, see change to update.sh

File: update.py [hunk 2]
```diff
@@ -750,8 +751,29 @@ def copy_static_html_sites(site, destdir):
             pass
 
 
-#######################################################################
+def check_imports():
+    '''check key imports work'''
+    import pkg_resources
+    requires = ["sphinx_rtd_theme>=0.1.8", "sphinxcontrib.vimeo>=0.0.1"]
```
Comment (TunaLobster) [inline update.py:757]: sphinxcontrib.vimeo does not exist. sphinxcontrib-vimeo does. However, neither are installed by the setup script any more. See #4195.

**Comment by TunaLobster** [comment]
Checking for the python modules shouldn't be necessary. Sphinx fails to attempt a build if a module isn't present. The real trick is modules with the same name but different behavior. There have been multiple sphinxcontrib-youtube extensions through the years. Checking the version and other details of the module installed in the current environment is more important. But in order to do that we need to lock down the versions used in the python environment. Future PR though.

**Comment by tridge** [comment]
> Checking for the python modules shouldn't be necessary. Sphinx fails to attempt a build if a module isn't present.

its a lot less confusing to give a clear error at startup of update.py than fail deep in the build after thousands of other lines of output

**Comment by tridge** [comment]
> The real trick is modules with the same name but different behavior

I've changed the install to use ArduPilot forks of the modules

**Pull request merged**